python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py

```diff
@@ -41,24 +48,78 @@ class LightGBMTrainer(Trainer):
             dataset. All datasets will be transformed by the ``preprocessor`` if
             one is provided. All non-training datasets will be used as separate
             validation sets, each reporting a separate metric.
-        lightgbm_config: LightGBM training parameters. Refer to
-            `LightGBM documentation <https://lightgbm.readthedocs.io/en\
-/latest/Parameters.html>`_ for a list of possible parameters.
+        lightgbm_config: LightGBM training parameters passed to ``lightgbm.train()``.
+            Refer to `LightGBM documentation <https://lightgbm.readthedocs.io>`_
+            for a list of possible parameters. Must contain a "params" key.
+            You can also pass a "dmatrix_params" key, which should be a dict
+            of ``dataset name:dict of kwargs`` passed to respective
+            :class:`xgboost_ray.RayDMatrix` initializations.
```

Comment:
Does it make sense to separate the DMatrix config into its own arg, rather than combining everything into a single config?

Reply:
To me, if we are going to have a particular schema, then I think it would make more sense to do this at the arg level instead of the dict level.